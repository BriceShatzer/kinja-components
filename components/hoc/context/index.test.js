// @flow

import * as React from 'react';
import { mount } from 'enzyme';

import { KinjaMetaProvider, withKinjaMeta } from './';
import type { KinjaMetaInjectedProps } from './';

describe('<KinjaMetaProvider />', () => {
	it('passes kinja metadata and preserves component props', () => {
		const BaseComponent = (props: {customContent: string} & KinjaMetaInjectedProps) => (
			<div>
				<h1>{props.kinjaMeta.config && props.kinjaMeta.config.insetPrefix}</h1>
				<p>{props.customContent}</p>
			</div>
		);

		const BaseComponentWithMetaData = withKinjaMeta(BaseComponent);

		const MiddleComponent = () => (
			<div>
				<span>Don&apos;t mind me I&apos;m just here doing nothing</span>
				<BaseComponentWithMetaData customContent={'World!'} />
			</div>
		);

		const providerProps = {
			config: {
				insetPrefix: 'Hello'
			}
		};

		const wrapper = mount(
			<KinjaMetaProvider value={providerProps} >
				<MiddleComponent />
			</KinjaMetaProvider>
		);

		expect(wrapper.getDOMNode()).toMatchSnapshot();
	});
});
