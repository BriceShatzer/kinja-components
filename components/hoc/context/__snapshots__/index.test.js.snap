// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KinjaMetaProvider /> passes kinja metadata and preserves component props 1`] = `
<div>
  <span>
    Don't mind me I'm just here doing nothing
  </span>
  <div>
    <h1>
      Hello
    </h1>
    <p>
      World!
    </p>
  </div>
</div>
`;
